subprojects/antlr/src/main/java/org/gradle/api/plugins/antlr/AntlrTask.java

```diff
@@ -271,16 +281,27 @@ public void setSource(Object source) {
     }
 
     /**
-     * Returns the source for this task, after the include and exclude patterns have been applied. Ignores source files which do not exist.
-     *
-     * @return The source.
+     * {@inheritDoc}
      */
     @Override
-    @PathSensitive(PathSensitivity.RELATIVE)
+    @Internal
     public FileTree getSource() {
         return super.getSource();
     }
 
+    /**
+     * The sources for incremental change detection.
+     *
+     * @since 6.0
+     */
+    @Incubating
+    @SkipWhenEmpty
+    @PathSensitive(PathSensitivity.RELATIVE)
+    @InputFiles
+    protected FileCollection getStableSources() {
+        return stableSources;
+    }
```

Comment:
Can't we add this to `SourceTask` somehow?

Reply:
We could, though I think we should aim for a better solution. This is currently only a stop-gap measure until we have a better file collection available.